apps/launch/components/MyStakingContracts/Create/useFieldRules.tsx

```diff
@@ -21,96 +21,102 @@ const getGenericFieldRules = (label: string) => [
 ];
 
 type StakingDepositRules = { [K in keyof FormValues]: { rules: Rule[] | undefined } };
-export const useFieldRules = (): StakingDepositRules => {
+export const useFieldRules = (
+  form: FormInstance,
+): {
+  onValuesChange: (changedValues: Record<string, unknown>) => void;
+  rulesConfig: StakingDepositRules;
+} => {
   const { data: numServicesLimit } = useNumServicesLimit();
   const { data: minStakingDepositLimit } = useMinStakingDepositLimit();
   const { data: timeForEmissionsLimit } = useTimeForEmissionsLimit();
   const { data: apyLimit } = useApyLimit();
 
+  const onValuesChange = (changedValues: Record<string, unknown>) => {
+    // The rewardsPerSecond validation depends on minStakingDeposit value
+    // Manually trigger it's validation when minStakingDeposit changes
+    if ('minStakingDeposit' in changedValues) {
+      form.validateFields(['rewardsPerSecond']);
+    }
+  };
+
   return {
-    contractName: { rules: getGenericFieldRules(FieldConfig.contractName.name) },
-    description: { rules: getGenericFieldRules(FieldConfig.description.name) },
-    maxNumServices: {
-      rules: [
-        ...getGenericFieldRules(FieldConfig.maxNumServices.name),
-        {
-          type: 'number',
-          min: 1,
-          max: numServicesLimit,
-          message: `Maximum number of staked agents must be at least 1 and at most ${numServicesLimit}`,
-        },
-      ],
-    },
-    rewardsPerSecond: {
-      rules: [
-        ...getGenericFieldRules(FieldConfig.rewardsPerSecond.name),
-        {
-          // BE validation from https://github.com/valory-xyz/autonolas-registries/blob/main/contracts/staking/StakingVerifier
-          validator: async (_, value) => {
-            if (!minStakingDepositLimit) return Promise.resolve();
-            if (!apyLimit) return Promise.resolve();
+    onValuesChange,
+    rulesConfig: {
+      contractName: { rules: getGenericFieldRules(FieldConfig.contractName.name) },
+      description: { rules: getGenericFieldRules(FieldConfig.description.name) },
+      maxNumServices: {
+        rules: [
+          ...getGenericFieldRules(FieldConfig.maxNumServices.name),
+          {
+            type: 'number',
+            min: 1,
+            max: numServicesLimit,
+            message: `Maximum number of staked agents must be at least 1 and at most ${numServicesLimit}`,
+          },
+        ],
+      },
+      rewardsPerSecond: {
+        rules: [
+          ...getGenericFieldRules(FieldConfig.rewardsPerSecond.name),
+          {
+            // BE validation from https://github.com/valory-xyz/autonolas-registries/blob/main/contracts/staking/StakingVerifier
+            validator: async (_, value) => {
+              if (!minStakingDepositLimit) return Promise.resolve();
+              if (!apyLimit) return Promise.resolve();
 
-            const rewardsPerYear = value * ONE_YEAR;
-            const apy = (rewardsPerYear * 1e18) / 1000;
```

Comment:
the `1000` was hardcoded preventing new contracts from being created. We should use the current `minStakingDeposit` value from the form instead